### PR TITLE
Updated Matrix API page with new class names and code

### LINKED ISSUE #356: Adjust Mapbox Java Service class names and code samples
Need to find and change the names of updated classes within Mapbox Java Services. For example,     https://www.mapbox.com/android-docs/mapbox-services/overview/directions-matrix/
 still has `DirectionsMatrixResponse` in the code example when the class is actually now named `MatrixResponse`. Off the top of my head, I know that similar changes are needed for the Optimization API.

### PR DESCRIPTION
**Pull request by langsmith**
Resolves #356 by updating text and code for the Matrix API page.

### CODE REVIEW

**Comment by langsmith** [comment]
Currently blocked by work mentioned in https://github.com/mapbox/android-docs/pull/375#discussion_r176138002

File: src/pages/mapbox-services/overview/directions-matrix.md [hunk 1]
```diff
@@ -1,46 +1,56 @@
...
+  public void onResponse(Call<MatrixResponse> call,
+    Response<MatrixResponse> response) {
+    
+    }
   }
 
   @Override
-  public void onFailure(Call<DirectionsMatrixResponse> call, Throwable t) {
+  public void onFailure(Call<MatrixResponse> call, Throwable throwable) {
+
```
Comment (cammace) [inline src/pages/mapbox-services/overview/directions-matrix.md]: nit extra line

Comment (langsmith) [reply src/pages/mapbox-services/overview/directions-matrix.md]: Extra line removed

**Comment by langsmith** [comment]
Anything else @cammace @colleenmcginnis ?

**Pull request merged**